src/Engines/Nunjucks.js

```diff
@@ -121,7 +122,35 @@ export default class Nunjucks extends TemplateEngine {
 
 	addFilters(filters, isAsync) {
 		for (let name in filters) {
-			this.njkEnv.addFilter(name, Nunjucks.wrapFilter(name, filters[name]), isAsync);
+			let callback = filters[name];
+			if (!isAsync) {
+				/** @this {any} */
+				callback = function (...args) {
+					// Note that `callback` is already a function as the `#add` method throws an error if not.
+					let ret = filters[name].call(this, ...args);
+					if (ret instanceof Promise) {
+						throw new Error(
+							`Nunjucks *is* async-friendly with \`addFilter("${name}", async function() {})\` but you need to supply an \`async function\`. You returned a promise from \`addFilter("${name}", function() {})\`. Alternatively, use the \`addAsyncFilter("${name}")\` configuration API method.`,
+						);
+					}
+					return ret;
+				}
+			}
+			else if (isAsync) {
```

Comment:
I could be wrong but it doesn't look like you need an `else if` because `isAsync` is Boolean already.